packages/pages/src/dev/server/server.ts

```diff
@@ -8,9 +8,11 @@ import { indexPage } from "./middleware/indexPage.js";
 import { generateTestData } from "./ssr/generateTestData.js";
 import { ProjectStructure } from "../../common/src/project/structure.js";
 import { finalSlashRedirect } from "./middleware/finalSlashRedirect.js";
+import { serverRenderSlugRoute } from "./middleware/serverRenderSlugRoute.js";
 
 export const createServer = async (
   dynamicGenerateData: boolean,
```

Comment:
Currently, if `dynamicGenerateData = true`, I believe this would call generateTestData(..), and then proceed to call generateTestDataForSlug / generateTestDataForPage on each new url path. I think the logic should be updated such that we either call `generateTestData` once on createServer OR call generateTestDataForSlug / generateTestDataForPage on each new route request?

Reply:
I believe we can't call generateTestData once on createServer because we don't know what URLs they'll want to test yet, and therefore don't know what slugs to call generateTestData for.

I think calling generateTestDataForPage once per new route request makes sense! You're essentially asking for caching, right? I imagine we'd want some way for user's to "bust" the cache and some time limit for cached generateTestData calls, say if a user wants to add new data to their Knowledge Graph and then test it in Pages. I think this is out of scope here though.

Reply:
I'm not sure if I follow, but we currently do [call `generateTestData()` in `createServer`](https://github.com/yext/pages/blob/dev/prod-urls-in-dev/packages/pages/src/dev/server/server.ts#L50) if `dynamicGenerateData` is true.

What I meant is, in this PR, if `dynamicGenerateData` is true (by default) and if user use `prod-url` flag, then we would generate 10 test data files randomly with `generateTestData()`, AND call `generateTestDataForSlug` for each time we go to an entity page. I think we should update the logic to NOT run that generate 10 test data files when `prod-url` is true, since that's not necessary. Similarly with `generateTestDataForPage`, if that is going to be executed, then we should need to call `generateTestData()`.

Reply:
I'm not sure why generateTestData gets called in create server only if dynamicGenerateData is true, I asked Duval on Slack

Reply:
Duval says this is to populate the index page

Reply:
if the user specifies --local we assume they have localData already

Reply:
gonna leave it like this for now if that's ok! we can definitely revisit